runtimes/runtimes/util/telemetryLspServer.ts

```diff
@@ -0,0 +1,37 @@
+import { Connection, InitializeParams, InitializeResult } from 'vscode-languageserver/node'
+import { Encoding } from '../encoding'
+import { Logging } from '../../server-interface/logging'
+import { LspServer } from '../lsp/router/lspServer'
+import { OperationalTelemetryProvider, TelemetryStatus } from '../operational-telemetry/operational-telemetry'
+
+export function getTelmetryLspServer(lspConnection: Connection, encoding: Encoding, logging: Logging): LspServer {
+    const lspServer = new LspServer(lspConnection, encoding, logging)
+    lspServer.setInitializeHandler(async (params: InitializeParams): Promise<InitializeResult> => {
+        const optOut: boolean = params.initializationOptions?.telemetryOptOut ?? false
```

Comment:
Should we differentiate between `undefined` and `false`. Imagine clients who get new version of LSP server and do not set `telemetryOptOut` yet. I do not think we want to increment major version because of this.

Reply:
right, changed it to opted-out if not provided.